lib/matplotlib/colors.py

```diff
@@ -2874,6 +2876,14 @@ def __call__(self, value, clip=None):
                 clip = self.clip
             if clip:
                 value = np.clip(value, self.vmin, self.vmax)
+
+            if norm_before_trf:
+                t_value = value - self.vmin
+                t_value /= (self.vmax - self.vmin)
+                t_value = self._trf.transform(t_value).reshape(np.shape(t_value))
+                t_value = np.ma.masked_invalid(t_value, copy=False)
+                return t_value[0] if is_scalar else t_value
```

Comment:
`t_value` should be used to represent the **transformed** value.

```suggestion
            if norm_before_trf:
                value -= self.vmin
                value /= (self.vmax - self.vmin)
                t_value = self._trf.transform(value).reshape(np.shape(value))
                t_value = np.ma.masked_invalid(t_value, copy=False)
                return t_value[0] if is_scalar else t_value
```